src/main/java/net/dv8tion/jda/internal/utils/cache/MemberCacheViewImpl.java

```diff
@@ -92,17 +92,17 @@ public List<Member> getElementsWithRoles(@Nonnull Role... roles)
     public List<Member> getElementsWithRoles(@Nonnull Collection<Role> roles)
     {
         Checks.noneNull(roles, "Roles");
-        if (isEmpty())
-            return Collections.emptyList();
```

Comment:
Could you explain why this has been moved after `rolesWithoutPublicRoleSet` has been computed?